addons/bus/models/bus_listener_mixin.py

```diff
@@ -0,0 +1,20 @@
+from odoo import models, api
+
+
+class BusListenerMixin(models.AbstractModel):
+    """Allow sending messages related to the current model via as a bus.bus channel.
+
+    The model needs to be allowed as a valid channel for the bus in `_build_bus_channel_list`.
+    """
+    _name = 'bus.listener.mixin'
+    _description = "Can send messages via bus.bus"
+
+    @api.model
+    def _bus_send(self, notification_type, message, subchannel=None):
+        """Send a notification to the webclient."""
+        for record in self:
+            channel = record._bus_channel() if subchannel is None else (record._bus_channel(), subchannel)
+            self.env['bus.bus']._add_to_queue(channel, notification_type, message)
+
+    def _bus_channel(self):
+        return self
```

Comment:
Note: this makes it harder to run a ci/check based on regex as the weak part now becomes the result of this function (checking that it's not returning an unsafe str for example).

Reply:
Maybe we would want to enforce that the returned value of `_bus_channel` is a recordset? Does that currently hold true for all cases?

Reply:
> Does that currently hold true for all cases?

Yes

Reply:
I added `assert isinstance(main_channel, models.Model)` in `_bus_send`